buildpack.md

```diff
@@ -476,35 +476,62 @@ The purpose of launch is to modify the running app environment using app-provide
 **GIVEN:**
 - An OCI image exported by the lifecycle,
 - An optional process type specified by `CNB_PROCESS_TYPE`, and
-- Bash version 3 or greater,
-
-When the OCI image is launched,
+- Bash version 3 or greater, if needed,
 
-1. The lifecycle MUST source each file in each `<layers>/<layer>/profile.d` directory using the same Bash shell process,
-   1. Firstly, in order of `/bin/build` execution used to construct the OCI image.
-   2. Secondly, in alphabetically ascending order by layer directory name.
-   3. Thirdly, in alphabetically ascending order by file name.
+First, the lifecycle MUST locate a start command and choose an execution strategy.
 
-2. In the Bash shell process used to source the `profile.d` scripts, the lifecycle MUST source `<app>/.profile` if it is present.
+To locate a start command,
 
-3. If `CMD` in the container configuration is not empty, the lifecycle MUST join each argument with a space and execute the resulting command in the container using the Bash shell process used to source the `profile.d` scripts.
+1. **IF** `CMD` in the container configuration is not empty,
+    **THEN** the value of `CMD` is chosen as the start command.
 
-4. If `CMD` in the container configuration is empty,
+2. **IF** `CMD` in the container configuration is empty,
    1. **IF** the `CNB_PROCESS_TYPE` environment variable is set,
       1. **IF** the value of `CNB_PROCESS_TYPE` corresponds to a process in `<layers>/launch.toml`, \
-         **THEN** the lifecycle MUST execute the corresponding command in the container using the Bash shell process used to source the `profile.d` scripts.
+         **THEN** the lifecycle MUST choose the corresponding process as the start command.
 
       2. **IF** the value of `CNB_PROCESS_TYPE` does not correspond to a process in `<layers>/launch.toml`, \
          **THEN** launch fails.
 
    2. **IF** the `CNB_PROCESS_TYPE` environment variable is not set,
       1. **IF** there is a process with a `web` process type in `<layers>/launch.toml`, \
-         **THEN** the lifecycle MUST execute the corresponding command in the container using the Bash shell process used to source the `profile.d` scripts.
+         **THEN** the lifecycle MUST choose the corresponding process as the start command.
 
       2. **IF** there is not a process with a `web` process type in `<layers>/launch.toml`, \
          **THEN** launch fails.
+         
+To choose an execution strategy,
+
+1. **IF** the value of `CMD` is chosen as the start command,
+   1. **IF** the length of the value of `CMD` is one,
+      **THEN** the lifecycle MUST invoke the value as a command using Bash
+
+   2. **IF** the length of the value of `CMD` is greater than one,
+      **THEN** the lifecycle MUST invoke the first entry using the `execve` syscall with subsequent entries as arguments.
```

Comment:
Does this mean that given the following:

* My stack has Bash
* My `CMD` is `myapp 42`

The lifecycle will invoke using `execve`? 

What are the implications of using `execve` in this case?

Reply:
It means `CMD [“myapp 42”]` will `execve` bash and behave as today. But `CMD [“myapp”, “42”]` will skip Bash (and thus profile.d/.profile) by `execve`ing `myapp` directly (with a `PATH` lookup first).

Reply:
Ah yea, the `CMD [“myapp”, “42”]` is what I was really asking about, sorry.

I'm concerned that will cause some unexpected results for users who have Bash, and simply try to set a CMD with multiple elements.

What does `docker run myimage myapp 42` become effectively? Would users need to do `docker run myimage "myapp 42"` or something?

Reply:
`docker run myimage myapp 42` would skip bash.
`docker run myimage "myapp 42"` would not skip bash.

I'm not sure how common it is to invoke `docker run` directly, but there's definitely potential to trip people up. Maybe we should introduce a `CNB_DIRECT_EXEC` or `CNB_SKIP_BASH` env var instead?

Reply:
What if setting the first argument to `-` triggered bash-less mode?
E.g.,
```
CMD [“-”, “myapp”, "42"]
```
```
docker run myimage - myapp 42
```

or maybe `=`?
E.g.,
```
CMD [“=”, “myapp”, "42"]
```
```
docker run myimage = myapp 42
```